app/Components/preferences/preferences.js

```diff
@@ -0,0 +1,76 @@
+
+const WebComponentBase = require('../WebComponentBase');
+
+const preferencesStore = require('../../data/preferencesStore');
+
+const { app } = require('electron').remote;
+
+module.exports = class Preferences extends WebComponentBase {
+    connectedCallback() {       
+        /**
+         * @type {ShadowRoot}
+         */
+        let shadow = this.attachShadow({ mode: 'open'});
+
+        this.applyHTML(shadow, __dirname, 'preferences.html');
+
+        shadow.querySelector('form')
+            .addEventListener('submit', this._onSubmit.bind(this));
+
+        const preferences = preferencesStore.getPreferences();
+        
+        shadow.querySelector('#runAtStartup').checked = preferences.runAtStartup;
+        shadow.querySelector('#waitTimeout').value = preferences.waitTimeout;
+        shadow.querySelector('#autoOpenReleasenotes').checked = preferences.autoOpenReleasenotes;
+    }
+
+    savePreferences() {
+        const formEl = this.shadowRoot.querySelector('form');
+
+        this._onSubmit(formEl);
+    }
+
+    _emitSaved() {
+        const e = new CustomEvent('preferences-saved', {
+            bubbles: true,
+            cancelable: true
+        });
+
+        this.dispatchEvent(e);
+    }
+
+    /**
+     * 
+     * @param {Event} e 
+     */
+    _onSubmit(form) {
+        const data = new FormData(form);
+        const runAtStartup = !!data.get('runAtStartup');
+        const waitTimeout = data.get('waitTimeout');
+        const autoOpenReleasenotes = !!data.get('autoOpenReleasenotes');
+
+        const preferences = preferencesStore.getPreferences();
+
+        this._manageStartUp(runAtStartup);
```

Comment:
Move this to after the save.

If save errors end up with start startup when the app reports its not.